src/phasorpy/utils.py

```diff
@@ -66,3 +73,157 @@ def number_threads(
     if max_threads is None:
         return num_threads
     return min(num_threads, max(max_threads, 1))
+
+
+def phasor_filter(
+    real: ArrayLike,
+    imag: ArrayLike,
+    /,
+    *,
+    method: str = 'median',
+    repeat: int = 1,
+    **kwargs: Any,
+) -> tuple[NDArray[Any], NDArray[Any]]:
+    """Apply a filter to phasor coordinates.
+
+    By default a median filter is applied to the real and imaginary
+    components of phasor coordinates once with a kernel size of 3
+    multiplied by the number of dimensions of the input arrays.
+
+    Parameters
+    ----------
+    real : array_like
+        Real component of phasor coordinates to be filtered.
+    imag : array_like
+        Imaginary component of phasor coordinates to be filtered.
+    method : str, optional
+        Method used for filtering:
+
+        - ``'median'``: Spatial median of phasor coordinates.
+
+    repeat : int, optional
+        The number of times to apply the median filter. Default is 1.
+    **kwargs
+        Optional arguments passed to :py:func:`scipy.ndimage.median_filter`.
+
+    Returns
+    -------
+    real : ndarray
+        Filtered real component of phasor coordinates.
+    imag : ndarray
+        Filtered imaginary component of phasor coordinates.
+
+    Raises
+    ------
+    ValueError
+        If the specified method is not supported.
+        The array shapes of `real` and `imag` do not match.
+        If `repeat` is less than 1.
+
+    Notes
+    -----
+    For now only the median filter method is implemented.
+    Additional filtering methods may be added in the future.
+
+    Examples
+    --------
+    Apply once a median filter with a kernel size of 3:
+
+    >>> phasor_filter(
+    ...     [[2, 2, 2], [1, 1, 1], [3, 3, 3]],
+    ...     [[4, 4, 4], [6, 6, 6], [5, 5, 5]],
+    ... )
+    (array([[2, 2, 2],
+            [2, 2, 2],
+            [3, 3, 3]]),
+    array([[4, 4, 4],
+            [5, 5, 5],
+            [5, 5, 5]]))
+
+    Apply 3 times a median filter with a kernel size of 3:
+
+    >>> phasor_filter(
+    ...     [[0, 0, 0], [5, 5, 5], [2, 2, 2]],
+    ...     [[3, 3, 3], [6, 6, 6], [4, 4, 4]],
+    ...     size=3,
+    ...     repeat=3,
+    ... )
+    (array([[0, 0, 0],
+            [2, 2, 2],
+            [2, 2, 2]]),
+    array([[3, 3, 3],
+            [4, 4, 4],
+            [4, 4, 4]]))
+
+    """
+    supported_methods = ['median']
```

Comment:
It's better not to duplicate this. Define `methods = {'median': _median_filter}` here and use it for both, `if method not in methods` and `return methods[method](...)`.